src/utils/filterFunction.ts

```diff
@@ -0,0 +1,95 @@
+                desiredPropertyTypes.forEach((propertyType: string) => {
+                    if (totalPropertyTypeAdded < maxResultsForPropertyTypes && (propertyTypeMatches || desiredPropertyTypes.includes('All') || property.Property_Category_Mailing.includes(propertyType))) {
+                        if (!propertyTypeMatches) {
+                            totalPropertyTypeAdded++
+                        }
+                        propertyTypeMatches = true
+                    }
+                })
+
+                if (!propertyTypeMatches) {
+                    desiredPropertyGroups.forEach((propertyGroup: string) => {
+                        if (totalPropertyGroupAdded < maxResultsForPropertyGroups && (propertyGroupMatches || desiredPropertyGroups.includes('All') || property.get('Property_Type_Portals').includes(propertyGroup))) {
+                            if (!propertyGroupMatches) {
+                                totalPropertyGroupAdded++
+                            }
+                            propertyGroupMatches = true
+                        }
+                    })
+                }
+
+                const canAddAsNeighbour = totalNeighboursAdded < maxNumNeigbours
+                const canAddBasedOnFilters = propertyGroupMatches || propertyTypeMatches
+                if (!canAddBasedOnFilters && canAddAsNeighbour) {
+                    totalNeighboursAdded++
+                }
+
+                if (canAddBasedOnFilters || canAddAsNeighbour) {
+                    const ownerData: any[] = []
+
+                    const parsedPropertyContacts = typeof property.Property_Contact === 'undefined' ? [] : JSON.parse(property.Property_Contact)
+                    parsedPropertyContacts.forEach((contact: any) => {
+                        contact.Contact_Type = 'Director'
+                        ownerData.push(contact)
+                    })
+
+                    const parsedPropertyOwners = typeof property.Property_Owners === 'undefined' || typeof property.Property_Owners === 'object' ? [] : JSON.parse(property.Property_Owners)
+
+                    parsedPropertyOwners.forEach((owner: any) => {
+                        owner.Contact_Type = 'Owner'
+                        ownerData.push(owner)
+                    })
+
+                    if (ownerData.length > 0) {
+                        property.owner_details = ownerData
+                        property.distance = propertyDistance.split('dist')[1]
+                        if (property.Managed === 'Yes' && managed === 'Yes') {
+                            matchedProperties.push(property)
+                            propertyMap[propertyAddress] = 'found'
+                        } else if (property.Managed === 'No' && managed === 'No') {
+                            matchedProperties.push(property)
+                            propertyMap[propertyAddress] = 'found'
+                        } else {
+                            matchedProperties.push(property)
```

Comment:
I don't think the logic is duplicated as the conditions are all different. Did you just want me to extract the repeated code into a function?

Reply:
My reading of the function is that it will always mark the property as matched due to the else statement.

Perhaps you want
```
if ((property.Managed === managed) || !managed) {
  matchedProperties.push(property)
  propertyMap[propertyAddress] = 'found'
}
```